onnxruntime/core/providers/cpu/ml/label_encoder.h

```diff
@@ -43,5 +43,66 @@ class LabelEncoder final : public OpKernel {
   int64_t default_int_;
 };
 
+template <typename TKey, typename TValue>
+class LabelEncoder_2 final : public OpKernel {
+ public:
+  LabelEncoder_2(const OpKernelInfo& info) : OpKernel(info) {
+    // Let the specialized member function to tell which fields to load.
+    InitializeSomeFields(info);
+
+    std::vector<TKey> keys;
+    std::vector<TValue> values;
+
+    ORT_ENFORCE(info.GetAttrs<TKey>(_key_field_name, keys).IsOK());
+    ORT_ENFORCE(info.GetAttrs<TValue>(_value_field_name, values).IsOK());
+
+    auto num_keys = keys.size();
+    auto num_values = values.size();
+    ORT_ENFORCE(num_keys == num_values,
+                "The ", _key_field_name, " and ", _value_field_name, " attribtues in LabelEncoder ",
+                "(name: ", info.node().Name(), ") must have the same length. ",
+                "However, the number of key is ", num_keys, " and the number of ",
+                "values is ", num_values, ".");
+
+    for (size_t i = 0; i < num_keys; ++i)
+      _map[keys[i]] = values[i];
+  }
+
+  Status Compute(OpKernelContext* context) const override {
+    const auto* tensor_pointer = context->Input<Tensor>(0);
+    if (tensor_pointer == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
+    const Tensor& X = *tensor_pointer;
+    const TensorShape& shape = X.Shape();
+    Tensor& Y = *context->Output(0, TensorShape(shape));
+
+    auto input = X.template DataAsSpan<TKey>();
+    auto output = Y.template MutableDataAsSpan<TValue>();
+
+    for (int64_t i = 0; i < shape.Size(); ++i) {
+      if (_map.find(input[i]) == _map.end())
```

Comment:
maybe reuse the result of find to avoid looking again in the map with ``.at(…)``